### PR TITLE
Communicate site creation edge cases to the users

### PR DESCRIPTION
**Pull request by aerosol**
### Changes

This PR introduces the following changes:
  - Domains with existing events cannot be created; either asynchronous deletion is in progress or support intervention is necessary.
image

  - I found user-facing code presenting a warning when site limit is reached, but it was never displayed (raw 4xx error page was returned). It is now shown when applicable:
image

  - The whole site creation process is wrapped with a single Postgres transaction
  - Some tests were naively misusing `Repo.exists?` so that it always evaluated to true - this has been fixed

### Tests
- [x] Automated tests have been added
- [ ] This PR does not require tests

### Changelog
- [ ] Entry has been added to changelog
- [x] This PR does not make a user-facing change

### Documentation
- [ ] [Docs](https://github.com/plausible/docs) have been updated
- [x] This change does not need a documentation update

### Dark mode
- [ ] The UI has been tested both in dark and light mode
- [x] This PR does not change the UI

### CODE REVIEW

File: lib/plausible/sites.ex [hunk 1]
```diff
@@ -166,4 +183,22 @@ defmodule Plausible.Sites do
...
   end
+
+  defp check_for_existing_events(domain, params) do
+    if has_events?(domain) do
+      Sentry.capture_message("Refused to create a site with existing events",
+        extra: %{params: params}
+      )
+
+      [
+        domain: """
```
Comment (ukutaht) [inline lib/plausible/sites.ex:194]: I'm not sure if we need to expose the inner workings of async deletes to users. How about showing the same error message as when the site already exists in the postgres database here?

Screenshot 2023-01-18 at 12 08 32

We would still get the error message sent to devs through Sentry so we can figure it out on our end.

cc @metmarkosaric what do you think?

File: lib/plausible_web/templates/site/new.html.eex [hunk 2]
```diff
@@ -16,7 +16,7 @@
             </h3>
             <div class="mt-2 text-sm text-yellow-700 dark:text-yellow-300">
               <p>
-              Your account is limited to <%= @site_limit %> sites. Please contact hello@plausible.io to add more sites.
+              Your account is limited to <%= @site_limit %> sites. Please contact support to add more sites.
```
Comment (ukutaht) [inline lib/plausible_web/templates/site/new.html.eex:19]: I think 'contact support' with no link or email address is not the best experience for the customer. They'll have to go hunting for how to contact support in the footer or something like this.

Self-hosted installations don't have a site limit so it's no problem to show our contact email address here. I think it's better UX to revert this or link the words 'contact support' to https://plausible.io/contact

Comment (aerosol) [reply lib/plausible_web/templates/site/new.html.eex:19]: Cheers, good catch, I initially meant to include https://plausible.io/contact link right under the "create site" modal. I suppose clickable error message text is preferred? 

I didn't like that there's a single screen technically advising using two different emails support@plausible.io and hello@plausible.io depending on the error.

File: lib/plausible/sites.ex (hunk 1 above)
Comment (aerosol) [reply lib/plausible/sites.ex:194]: Agreed that's a good call; no need to be very specific here as we'll still keep track of it internally.

Comment (metmarkosaric) [reply lib/plausible/sites.ex:194]: yeah, i agree. i mentioned the same thing yesterday on the basecamp thread

Comment (aerosol) [reply lib/plausible/sites.ex]: https://github.com/plausible/analytics/pull/2605/commits/eaefca7d5c22ddd90ced43c84e3f403419594e22

File: lib/plausible_web/templates/site/new.html.eex (hunk 2 above)
Comment (aerosol) [reply lib/plausible_web/templates/site/new.html.eex]: Included the e-mail back again in https://github.com/plausible/analytics/pull/2605/commits/54b4975038dbc6028c88d2b31205d303a1fd868f

Is that good enough?

Comment (metmarkosaric) [reply lib/plausible_web/templates/site/new.html.eex]: looks fine to me @aerosol. we get all emails sent to both of these addresses in Help Scout but also happy if you want to pick one and standardize all the messages in the app. 

we typically link to https://plausible.io/contact as there people can find links to the docs/faq. this helps reduce the number of emails we got and works for more generic error messages. when the error is very specific such as this case where people actually need to contact us, it's best experience to just give them the email address directly

**Review by vinibrsl — APPROVED** [review]

**Pull request merged**